File: Module1/Exercises/Taller3/Python/punto2/src/tv.py
class TV:
    _numTV = 0

    def __init__(self,marca,control,canal=1,precio=500,estado=False,volumen=1):
        self.marca = marca
        self.canal = canal
        self.precio = precio
        self.estado = estado
        self.volumen = volumen
        self.control = control

        TV._numTV += 1

    def getEstado(self):
        return self.estado

    def setMarca(self,marca):
        self.marca = marca

    def getMarca(self):
        return self.marca
        
    def setcontrol(self,control):
        self.control = control

    def getControl(self):
        return self.control

    def setPrecio(self,precio):
        self.precio = precio

    def getPrecio(self):
        return self.precio

    def setVolumen(self,volumen):
        if 0 <= volumen <= 7 and self.estado:
            self.volumen = volumen

    def getVolumen(self):
        return self.volumen

    def setCanal(self,canal):
        if 1 <= canal <= 120 and self.estado:
            self.canal = canal

    def getCanal(self):
        return self.canal

    def canalUp(self):
        if self.getEstado() and 1 < self.canal < 120:
         self.canal += 1

    def canalDown(self):
        if self.getEstado() and 1 < self.canal < 120:
            self.canal -= 1
    
    def volumenUp(self):
        if self.getEstado() and 0 < self.volumen < 7:
            self.volumen += 1

    def volumenDown(self):
        if self.getEstado() and 0 < self.volumen < 7:
            self.volumen -= 1

    @classmethod
    def getCantidad(cls):
        return cls._numTV

    

        

File: Module1/Classes/Class3/Taller3/Java/punto2/TV.java

@SuppressWarnings("unused")
public class TV {

    private Marca marca;
    private int canal = 1;
    private int precio = 500;
    private boolean estado;
    private int volumen = 1;
    private Control control;
    private static int numTV;

    public TV(Marca marca, boolean estado){

        this.marca = marca;
        this.estado = false;
        numTV++;
    }

    private void setMarca(Marca marca){
        this.marca = marca;

    }

    private Marca getMarca(){
        return this.marca;
    }

    void setControl(Control control){
        this.control = control;
    }

    private Control getControl(){
        return this.control;
    }

    private void setPrecio(int precio){
        this.precio = precio;
    }

    private int getPrecio(){
        return this.precio;
    }

    private void setVolumen(int volumen){

        if(estado && this.volumen <= 7 && this.volumen >= 0){
            this.volumen = volumen;
        }
    }

    int getVolumen(){
        return volumen;
    }

    void setCanal(int canal){

        if(estado && this.canal >= 1 && this.canal <= 120){
            this.canal = canal;
        }
    }

    int getCanal(){
        return this.canal;
    }

    private static int getNumTV(){
        return numTV;
    }

    void turnOff(){
        this.estado = false;
    }

    public void turnOn(){
        this.estado = true;

    }

    private boolean getEstado(){
        return this.estado;
    }

    void canalUp(){

        if(estado && this.canal < 120){
            this.canal++;
        }
    }

    void canalDown(){

        if(estado && this.canal > 1){
            this.canal--;
        }
    }

    void volumenUp(){

        if(estado && this.volumen > 0 && this.volumen < 7){
            this.volumen++;
        }

    
      
    }

    void volumenDown(){

        if(estado && this.volumen < 7 && this.volumen > 0){
            this.volumen--;
        }
    }
    
}
